packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -520,30 +520,46 @@ class CupertinoActionSheet extends StatelessWidget {
   /// short.
   final ScrollController? messageScrollController;
 
-  ScrollController get _effectiveMessageScrollController =>
-    messageScrollController ?? ScrollController();
-
   /// A scroll controller that can be used to control the scrolling of the
   /// [actions] in the action sheet.
   ///
   /// This attribute is typically not needed.
   final ScrollController? actionScrollController;
 
-  ScrollController get _effectiveActionScrollController =>
-    actionScrollController ?? ScrollController();
-
   /// The optional cancel button that is grouped separately from the other
   /// actions.
   ///
   /// Typically this is an [CupertinoActionSheetAction] widget.
   final Widget? cancelButton;
 
+  @override
+  State<CupertinoActionSheet> createState() => _CupertinoActionSheetState();
+}
+
+class _CupertinoActionSheetState extends State<CupertinoActionSheet> {
+  late final ScrollController _effectiveMessageScrollController =
```

Comment:
This won't work if the widget first built with widget.messageScrollController == null and then rebuild with widget.messageScrollController != null; once this is assigned it will not be updated later.

I think you will need a variable to hold internal `ScrollController` and have a getter for `_effectiveMessageScrollController`